### PR TITLE
Tauri Autostart intregration

### PR DESCRIPTION
**Pull request by AdenMGB**
### Autostart using the Tauri Autostart plugin

1. Added plugin dependency in cargo.toml
2. Added setting toggle for Autostart in settings
3. Created autostart.rs
4. Edited lib.rs

Plugin page
https://v2.tauri.app/plugin/autostart/

### CODE REVIEW

File: src-tauri/src/lib.rs [hunk 1]
```diff
@@ -234,10 +236,16 @@ pub fn run() {
...
             launch_game,
-            kill_game
+            kill_game,
+            toggle_autostart,
+            get_autostart_enabled,
         ])
         .plugin(tauri_plugin_shell::init())
         .plugin(tauri_plugin_dialog::init())
+        .plugin(tauri_plugin_autostart::init(
+            tauri_plugin_autostart::MacosLauncher::LaunchAgent,
```
Comment (DecDuck) [inline src-tauri/src/lib.rs:246]: MacOS launcher?

File: src-tauri/src/autostart.rs
Comment (DecDuck) [inline src-tauri/src/autostart.rs:1]: Can we store the state in the DB, and then sync it up on startup/on change?

File: pages/settings/index.vue
Comment (DecDuck) [inline pages/settings/index.vue:1]: Haven't looked at the design yet

**Review by DecDuck — CHANGES_REQUESTED** [review]

File: src-tauri/src/lib.rs (hunk 1 above)
Comment (AdenMGB) [reply src-tauri/src/lib.rs:246]: @DecDuck 
image
future proofing, its just the name they used in the documentation

https://v2.tauri.app/plugin/autostart/

Comment (AdenMGB) [reply src-tauri/src/lib.rs:246]: i can remove it i guess?

File: pages/settings/index.vue
Comment (AdenMGB) [reply pages/settings/index.vue:1]: image
functional 👍

File: src-tauri/src/lib.rs (hunk 1 above)
Comment (DecDuck) [reply src-tauri/src/lib.rs:246]: Oh so it's not a MacOS specific thing, good.

File: pages/settings/index.vue
Comment (DecDuck) [reply pages/settings/index.vue:1]: Looks good!

File: src-tauri/src/autostart.rs
Comment (AdenMGB) [reply src-tauri/src/autostart.rs:1]: Resolved, committed the necessary files to enable DB synce.

**Review by DecDuck — APPROVED** [review]
Looks great!

**Comment by DecDuck** [comment]
Merged with squash commit, so GitHub didn't detect it. Check out 7a5c127fc8d9aaa54abad94559bffd2cb3f493b1

**Pull request closed without merging**